architecture-compose/src/androidLibMain/kotlin/com/splendo/kaluga/architecture/compose/viewModel/KalugaViewModelComposable.kt

```diff
@@ -0,0 +1,110 @@
+package com.splendo.kaluga.architecture.compose.viewModel
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelStore
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.lifecycle.viewmodel.compose.LocalViewModelStoreOwner
+import com.splendo.kaluga.architecture.viewmodel.BaseViewModel
+import org.koin.androidx.viewmodel.ViewModelOwner
+import org.koin.androidx.viewmodel.koin.getViewModel
+import org.koin.core.parameter.parametersOf
+import org.koin.java.KoinJavaComponent.getKoin
+
+@Composable
+inline fun <reified ViewModel : BaseViewModel> getViewModel(vararg params: Any?): ViewModel {
+    val owner = ViewModelOwner.from(LocalViewModelStoreOwner.current!!)
+    return getKoin().getViewModel(owner = { owner }) {
+        parametersOf(*params)
+    }
+}
+
+/**
+ * Composable which manages [viewModel] lifecycle and optionally adds it to local [ViewModelStore].
+ * @param viewModel view model to manage
+ * @param store whether to add the [viewModel] a local [ViewModelStore]. Use if the [viewModel]
+ * was created manually and is not located in Activity/Fragment [ViewModelStore].
+ * @param content content based on [viewModel]
+ */
+@Composable
+fun <ViewModel : BaseViewModel> ViewModelComposable(
+    viewModel: ViewModel,
+    store: Boolean = false,
```

Comment:
unsure on the parameter name. considering to scrape it in favour of `store(..)` so instead of 

```
val vm = remember { SomeVM() }
ViewModelComposable(vm, store = true)
```

you'd write
```
val vm = store { remember { SomeVM() } }
ViewModelComposable(vm)
```